### DISCUSSION TITLE
How to define the TypeScript type of a middleware, such as "log" middleware in README.md ?

### DISCUSSION THREAD

**Discussion by puxiao** [pmndrs/zustand#1515] [Q&A] [answered]
Although I have read 
`zustad/mddleware/combine.d.ts`
`zustad/mddleware/persist.d.ts`
...

em
It is very difficult to understand. 
Can someone explain?

**Comment by dai-shi** [comment, marked as answer]
Does typescript guide in docs help?
https://github.com/pmndrs/zustand/blob/820dbd35eabc36d0117e3ac9ee9c1cb36336e88e/docs/guides/typescript.md?plain=1#L239-L264

But, it's very very difficult. Not recommended in general.
